.github/workflows/tox.yml

```diff
@@ -0,0 +1,142 @@
+name: tox
+
+on:
+  create:  # is used for publishing to PyPI and TestPyPI
+    tags:  # any tag regardless of its name, no branches
+  push:  # only publishes pushes to the main branch to TestPyPI
+    branches:  # any branch but not tag
+      - >-
+        **
+    tags-ignore:
+      - >-
+        **
```

Comment:
I wonder if this structure will pick up force-pushes of tags, or only creations.

Also, you could do

```yaml
branches:
  - "**"
tags-ignore:
  - "**"
```

If you wanted it to be more compact.